### PR TITLE
passed compilation on ubuntu 22.04 with some minor changes

### PR DESCRIPTION
**Pull request by yinweijie**

### CODE REVIEW

File: src/OpenFOAM/CMakeLists.txt [hunk 1]
```diff
@@ -609,4 +609,14 @@ target_link_libraries(OpenFOAM PUBLIC Pstream)
 target_include_directories(OpenFOAM PUBLIC
   .
 )
+
+find_package(ZLIB REQUIRED)
```
Comment (hsorby) [inline src/OpenFOAM/CMakeLists.txt]: We have this here: https://github.com/openfoamdotcom/openfoam/blob/d29f00600d42321444580c3d49d167045e01bce4/CMakeLists.txt#L25C1-L25C28

File: src/OpenFOAM/CMakeLists.txt [hunk 2]
```diff
@@ -609,4 +609,14 @@ target_link_libraries(OpenFOAM PUBLIC Pstream)
...
+if(ZLIB_FOUND)
+  add_definitions(-DHAVE_LIBZ)
+endif()
```
Comment (hsorby) [inline src/OpenFOAM/CMakeLists.txt]: And this https://github.com/openfoamdotcom/openfoam/blob/d29f00600d42321444580c3d49d167045e01bce4/src/OpenFOAM/CMakeLists.txt#L598 should cover these lines.

File: src/OpenFOAM/CMakeLists.txt [hunk 3]
```diff
@@ -609,4 +609,14 @@ target_link_libraries(OpenFOAM PUBLIC Pstream)
...
+
+# link openfoam with zlib
+target_link_libraries(OpenFOAM PUBLIC ${ZLIB_LIBRARIES})
```
Comment (hsorby) [inline src/OpenFOAM/CMakeLists.txt]: We have a target_link_libraries call here: https://github.com/openfoamdotcom/openfoam/blob/d29f00600d42321444580c3d49d167045e01bce4/src/OpenFOAM/CMakeLists.txt#L608

Comment (hsorby) [reply src/OpenFOAM/CMakeLists.txt]: And we should be able to use `ZLIB::ZLIB` as the link library so that we don't need the target include directories.

**Review by hsorby — CHANGES_REQUESTED** [review]
Thanks for this, can you try out me suggestions and see if it still builds. If it does I think that would match with the rest of the project nicely.

**Comment by hsorby** [comment]
I am wondering how I managed to build without linking to the zlib library??

**Comment by yinweijie** [comment]
> Thanks for this, can you try out me suggestions and see if it still builds. If it does I think that would match with the rest of the project nicely.

I try to compile on my another desktop ubuntu 22 envrionment and it passed compilation without need to make changes to zlib target link(changes to [thirdparty/CMakeLists.txt "-DCMAKE_POSITION_INDEPENDENT_CODE=ON"](https://github.com/openfoamdotcom/openfoam/pull/1/files#diff-63349b1a36da563bdafd8d31c8f86eb55bc49e0c8daf68d7b817daace0ee09abR84) is still needed). 
Things are little strange. I'll try again on my laptop on Monday.

**Comment by yinweijie** [comment]
> Thanks for this, can you try out me suggestions and see if it still builds. If it does I think that would match with the rest of the project nicely.

image

By testing on my laptop ubuntu 22.04, I still get the error without target link with ZLIB::ZLIB as shown above. So target link to ZLIB::ZLIB is needed.

Here is the final revision I made to your code base:
https://github.com/openfoamdotcom/openfoam/pull/1/files

**Comment by yinweijie** [comment]
> I am wondering how I managed to build without linking to the zlib library??

That's strange to me also. I can successfully build without target link to ZLIB::ZLIB  on my another laptop ubuntu environment.

**Review by hsorby — APPROVED** [review]

**Pull request merged**